src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepository.scala

```diff
@@ -21,6 +21,14 @@ class ConsignmentRepository(db: Database) {
     db.run(query.result)
   }
 
+  def getConsignmentFiles(consignmentId: UUID): Future[Seq[UUID]] = {
+    val query = for {
+      (file, _) <- File.join(Consignment).on(_.consignmentid === _.consignmentid).filter(_._1.consignmentid === consignmentId)
```

Comment:
In answer to your question about whether you need the join - I suspect not, because you don't need anything from the consignment table.

I guess we really only need to know if there are any files for the consignment. I was going to suggest the Slick equivalent of a SQL `COUNT` query, but I've just found the `exists` function: https://stackoverflow.com/a/33628053/679924 Does that do what you need, since it avoids having a query that might return data for thousands of files?

Reply:
Yes, that's a nice function. Thanks